src/services/sanity.js

```diff
@@ -1374,6 +1374,77 @@ export async function fetchGenreLessons(brand, name, contentType, {
   return fetchSanity(query, true);
 }
 
+export async function fetchTopLevelParentId(railcontentId) {
+    const query = `*[railcontent_id == ${railcontentId}]{
+      railcontent_id,
+      'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+        railcontent_id,
+          'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+            railcontent_id,
+            'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+              railcontent_id,
+               'parents': *[^._id in child[]._ref && !(_id in path('drafts.**'))]{
+                  railcontent_id,               
+            } 
+          }
+        }
+      }
+    }`;
+    let response = await fetchSanity(query, false, {processNeedAccess: false});
+    if (!response) return null;
+    let currentLevel = response;
+    for (let i = 0; i < 4; i++) {
```

Comment:
I wonder if this `4` should be an optional param for the function, in case of different content types or pages requiring different levels of hierarchy. (We'd also need to build the query dynamically based on that number to make sure we have the right number of levels).

Reply:
I also remembered that Roxana has done more work on hierarchy lately and I wonder if we could just directly call something from the piece of content in Sanity?

Reply:
This is meant to pull the top entity, so I don't know if having a depth here makes sense.  There could be a function to pull to a certain depths but if you don't know where your at I'm not sure what the use cases for that would actually be.  If we need it in the future we could revamp these functions to do that.